src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/OsmPbfProcessor.java

```diff
@@ -625,6 +631,23 @@ private boolean isHighwayOrFerry(final Way way)
         return HighwayTag.isCoreWay(taggableWay) || RouteTag.isFerry(taggableWay);
     }
 
+    /**
+     * @param node
+     *            The node to test for
+     * @return True only if the node is completely outside of any boundary. In that case, it has
+     *         either no country (COUNTRY_MISSING) or its country is assigned using the nearest
+     *         neighbor logic.
+     */
+    private boolean isOutsideBoundary(final Node node)
+    {
+        final Geometry nodeLocation = JTS_POINT_CONVERTER.convert(new Location(
+                Latitude.degrees(node.getLatitude()), Longitude.degrees(node.getLongitude())));
+        final CountryCodeProperties countryCodeProperties = this.loadingOption
+                .getCountryBoundaryMap().getCountryCodeISO3(nodeLocation, false);
+        return ISOCountryTag.COUNTRY_MISSING.equals(countryCodeProperties.getIso3CountryCode())
+                || countryCodeProperties.usingNearestNeighbor();
```

Comment:
I am not sure if this is the right place to ask question about nearest neighbor logic but how does it work? 
Is it something like 

if current node does not have ISO country tag then 
get the country code for the nearest neighbor and assign it to current node?

Reply:
@snb3300 - that's right. If a feature is close to a country, but not quite within the border, then we will try to find the nearest country and assign its country code. We will add the nearest neighbor tag to make this case a clear distinction.